File: cpp/2089.find-target-indices-after-sorting-array.cpp
class Solution {
private:
    int left(const vector<int>& arr, int target) {
        int lo = 0;
        int hi = arr.size();
        int mid;

        while (lo < hi) {
            mid = (lo + hi) / 2;
            if (arr[mid] < target) {
                lo = mid + 1;
            } else {
                hi = mid;
            }
        }
        return lo;
    }

    int right(const vector<int>& arr, int target) {
        int lo = 0;
        int hi = arr.size();
        int mid;

        while (lo < hi) {
            mid = (lo + hi) / 2;
            if (arr[mid] <= target) {
                lo = mid + 1;
            } else {
                hi = mid;
            }
        }
        return lo;
    }
public:
    vector<int> targetIndices(vector<int>& nums, int target) {
        std::sort(nums.begin(), nums.end(), std::less<int>());
        int lo = left(nums, target);
        int hi = right(nums, target);
        vector<int> res;
        for (int i = lo; i < hi; i++) {
            res.push_back(i);
        }
        return res;
    }
};

File: golang/2089.find-target-indices-after-sorting-array.go
func left(arr []int, target int) int {
    lo, hi := 0, len(arr)
    var mid int
    for lo < hi {
        mid = (lo + hi) / 2
        if arr[mid] < target {
            lo = mid + 1
        } else {
            hi = mid
        }
    }
    return lo
}

func right(arr []int, target int) int {
    lo, hi := 0, len(arr)
    var mid int
    for lo < hi {
        mid = (lo + hi) / 2
        if arr[mid] <= target {
            lo = mid + 1
        } else {
            hi = mid
        }
    }
    return lo
}

func targetIndices(nums []int, target int) []int {
    sort.Ints(nums)
    lo, hi := left(nums, target), right(nums, target)
    res := make([]int, 0, 2)
    for i := lo; i < hi; i++ {
        res = append(res, i)
    }
    return res
}